File: backend/db/models/todo.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship

from db.database import Base


class Todo(Base):
    __tablename__ = "todos"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, index=True, nullable=False)
    description = Column(String)
    is_completed = Column(Boolean, default=False)
    due_date = Column(DateTime, nullable=True)
    priority = Column(Integer, default=1)  # default priority
    todo_list_id = Column(Integer, ForeignKey("todo_lists.id"))
    owner_id = Column(Integer, ForeignKey("users.id"), index=True)
    created_at = Column(DateTime, server_default=func.now(), nullable=False)
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now(), nullable=False)

    todo_list = relationship("TodoList", back_populates="todos")

    def __repr__(self):
        return f"<Todo(id={self.id}, name={self.title}, due_date={self.due_date}, priority={self.priority})>"

File: frontend/lib/to-do.ts
type ToDo = {
  id: string;
  title: string;
  description: string | null;
  due_date: string | null;
  priority: number | null;
  todo_list_id: string | null;
  is_completed: boolean | null;
  owner_id: number;
  created_at: string;
  updated_at: string;
};

export default ToDo;
